core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/impl/api/parse.kt

```diff
@@ -171,18 +171,23 @@ internal object Parsers : GlobalParserOptions {
         return null
     }
 
-    private fun String.parseDouble(format: NumberFormat) =
+    private val posixNumberFormat = NumberFormat.getInstance(Locale.forLanguageTag("C.UTF-8"))
+
+    private fun String.parseDouble(userNumberFormat: NumberFormat) =
         when (uppercase(Locale.getDefault())) {
             "NAN" -> Double.NaN
             "INF" -> Double.POSITIVE_INFINITY
             "-INF" -> Double.NEGATIVE_INFINITY
             "INFINITY" -> Double.POSITIVE_INFINITY
             "-INFINITY" -> Double.NEGATIVE_INFINITY
             else -> {
-                val parsePosition = ParsePosition(0)
-                val result: Double? = format.parse(this, parsePosition)?.toDouble()
-                if (parsePosition.index != this.length) null
-                else result
+                fun parseWithFormat(format: NumberFormat): Double? {
+                    val parsePosition = ParsePosition(0)
+                    val result: Double? = format.parse(this, parsePosition)?.toDouble()
+                    return if (parsePosition.index != this.length) null
+                    else result
+                }
+                parseWithFormat(userNumberFormat) ?: parseWithFormat(posixNumberFormat)
```

Comment:
I wonder if it's possible to misinterpret some value this way?
Right now, in `parse`, we use one parser for a whole column. If a parser fails to produce a value, then we go to the next one. But here you try to parse first with the locale-specific one, then with POSIX.
row 1 = can parse with locale specific
row 2 = cannot parse with locale specific, fallback to POSIX
But then, maybe the whole column is in the POSIX format and we should use fallback parser for row 1 as well? Can it be that row 1 can be parsed with first and second parser but yield different results?

Reply:
Do we consider that both formats are mixed up in one column? If yes, than it can.

For example (as it is shown in unit test), string like `"12,345"` would mean `12.345` in ru-RU locale and `12_345` in POSIX. On other side, string like `"12.345"` would be illegal for ru-RU parser but is often used in real life.

So if we have column like `columnOf("12.345", "67,890")` and Russian locale by default, it would be converted to values `12.345`, `67.89` in one case (parser per row) and to values `12.345`, `67_890` in another (parser per column).

Reply:
But actually, if we have column with comma as thousands group delimiter (as in POSIX), like `columnOf("12,345", "67,890")` but ru-RU locale by default, it would be misinterpret anyway.

I do not have experience with other locales. There can be other kinds of controversies, may be.

Reply:
Hm, i think we should consider such columns if they exist in real world data, but I have too little experience. 
For example,  
`columnOf("12.345", "67,890")` is `12.345, 67_890` or `12.345, 67.89`? 

> But actually, if we have column with comma as thousands group delimiter (as in POSIX), like columnOf("12,345", "67,890") but ru-RU locale by default, it would be misinterpret anyway.
Valid point. So people would have to specify the format explicitly

Reply:
> For example,
> `columnOf("12.345", "67,890")` is `12.345, 67_890` or `12.345, 67.89`?

In my case (dirty data or manual input without ple-validation) it should be parsed as `12.345`, `67.89`. Current dataframe version throws `TypeConversionException: Failed to convert '12.345' from kotlin.String to kotlin.Double` on Russian locale. And I tried to avoid this.

> Valid point. So people would have to specify the format explicitly

`DataFrame.readCSV` already has ParserOptions parameter. Should we add the same to `DataColumn.convertTo()`?

Reply:
Maybe `map { it.replace(",", ".").toDouble() }` could work for you? For now, i still think that per row parsers could create confusion. 

I like the general idea of a fallback to POSIX though. But with per column parsing, it won't solve your problem, right? So we need a use case where it's useful :)  

And yes, probably we need to have `locale` parameter in `DataColumn<String>.convetToDouble` and `Convert<>.toDouble` overloads, like `DataColumn<String>.convertToLocalDate` does

Reply:
I have not found how to make `Convert<>.toDouble` with custom parameter. Code like 
```
public fun DataColumn<String>.convertTo<Double>(locale: Locale? = null): DataColumn<Double> {
    return this.convertToDouble(locale)
}
```
does not work...

Other requested features have been implemented.